### PR TITLE
feat: Invoice sub client and methods

### PR DESCRIPTION
**Pull request by phemmylintry**

### CODE REVIEW

**Comment by phemmylintry** [comment]
#5 
#4

**Review by parisk — CHANGES_REQUESTED** [review]
Good job @phemmylintry 👍. Let's fix a few things and I think we can move on.

File: billit/utils/tax_utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+@dataclass
+class Tax:
+    taxId: int = 0
+    taxVatShow: int = 0
+
+    # return a dict
+    def as_dict(self):
+        return asdict(self)
```
[the remark this answers is not in the document]
Comment (phemmylintry) [reply billit/utils/tax_utils.py:11]: @parisk where will the asdcit() function be used then?

[the remark this answers is not in the document]
Comment (parisk) [reply billit/utils/tax_utils.py:11]: We should use it before passing data to `_handle_request`. Example:

```py
def create(self, customerId: int, sendMail: bool, ..., taxes: Tax, ...):
    data = {
        "customerId": customerId,
        "sendMail": sendMail,
        ...,
        "taxes": dataclasses.as_dict(taxes),
        ...,
    }
    return self.client._handle_request("POST", "/invoices", data=data)
```

File: billit/client.py [hunk 2]
```diff
@@ -65,3 +68,77 @@ def __init__(self, client: Client):
...
+            "invoiceTypeId": invoiceTypeId,
+            "isPaid": isPaid,
+            "mydataInvoiceType": mydataInvoiceType,
+            "taxes": dataclass.asdict(taxes),
+            "products": products,
+            "tags": tags,
+            "mydataPayment": mydataPayment,
+            "mailOptions": mailOptions,
+            "reminder": reminder,
+        }
```
Comment (parisk) [inline billit/client.py:107]: There is one last thing we need to take care here. We need to have Python arguments in snake case[^1], as this is the official coding style proposed by PEP8[^2] (e.g. use `customer_id` in Python, as described in #4, and convert it to `customerId` before sending to the API).

In order to make this easier and DRY between `create` and `update`, I suggest doing something like the following:

```py
class Invoices(SubClient):
    _args_api_mappings = {
        "customer_id": "customerId",
        "send_mail": "sendMail",
        # ...
    }
    
    def create(self, customer_id: int, send_mail: bool, ...):
        data = {
            self._args_api_mappings["customer_id"]: customer_id,
            self._args_api_mappings["send_mail"]: send_mail,
            # ...
        }
        return self.client._handle_request("POST", "/invoices", data=data)
    
    # ...
    
    def update(self, customer_id: int, send_mail: bool, ...):
        data = {
            self._args_api_mappings["customer_id"]: customer_id,
            self._args_api_mappings["send_mail"]: send_mail,
            # ...
        }
        return self.client._handle_request("PUT", "/invoices", data=data)
```

[^1]: https://en.wikipedia.org/wiki/Snake_case
[^2]: https://peps.python.org/pep-0008/#function-and-variable-names

**Review by parisk — CHANGES_REQUESTED** [review]
Great job @phemmylintry. We are almost there.  Let's make arguments PEP8 compliant and then we can merge this.

File: billit/client.py [hunk 3]
```diff
@@ -65,3 +69,94 @@ def __init__(self, client: Client):
...
+        return self.client._handle_request("GET", "/invoices")
+
+    def create(
+        self,
+        customer_id: int,
+        send_mail: bool,
+        exclude_mydata: bool,
+        invoice_date: str,
+        invoice_type_id: int,
+        isPaid: bool,
```
Comment (parisk) [inline billit/client.py:101]:
```suggestion
        is_paid: bool,
```

**Review by parisk — APPROVED** [review]
Great job! 👏 👏 👏 

LGTM ✌️

**Pull request merged**